src/wrapper/Plumed.h

```diff
@@ -2874,28 +2947,274 @@ Plumed(Plumed&&p)__PLUMED_WRAPPER_CXX_NOEXCEPT :
     plumed_cmd_cxx(main,key);
   }
 
+#if __cplusplus > 199711L && __PLUMED_WRAPPER_CXX_DETECT_SHAPES
+
+private:
+
+  /// Internal tool to convert initializer_list to shape
+  /// This is just taking an initializer list and making a std::array
+  std::array<std::size_t,5>  make_shape(std::initializer_list<std::size_t> shape) {
+    if(shape.size()>4) throw Plumed::ExceptionTypeError("Maximum shape size is 4");
+    std::array<std::size_t,5> shape_;
+    unsigned j=0;
+    for(auto i : shape) {
+      shape_[j]=i;
+      j++;
+    }
+    shape_[j]=0;
+    return shape_;
+  }
+
+  /// Internal utility to append a shape.
+  /// Create a new shape where newindex has been appended to the last non zero element.
+  std::array<std::size_t,5> append_size(std::size_t* shape,std::size_t newindex) {
+    std::array<std::size_t,5> shape_;
+    unsigned i;
+    for(i=0; i<4; i++) {
+      shape_[i]=shape[i];
+      if(shape[i]==0) break;
+    }
```

Comment:
In theory this is documented to be a zero-terminated array. However I agree, it should be checked at the entry point to avoid user errors and raise an exception instead!